airflow/providers/amazon/aws/transfers/redshift_to_s3.py

```diff
@@ -106,24 +108,30 @@ def __init__(  # pylint: disable=too-many-arguments
 
     def execute(self, context):
         postgres_hook = PostgresHook(postgres_conn_id=self.redshift_conn_id)
-        s3_hook = S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
 
-        credentials = s3_hook.get_credentials()
+        if self.aws_iam_role is not None:
+            access_options = "IAM_ROLE '{}'".format(self.aws_iam_role)
+        else:
+            s3_hook = S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
+            credentials = s3_hook.get_credentials()
```

Comment:
I wonder if we should look at the `aws_iam_role` field of extras - http://airflow.apache.org/docs/stable/howto/connection/aws.html as well?